Modules/Devices/src/Devices.Application/PushNotifications/Commands/UpdateDeviceRegistration/Handler.cs

```diff
@@ -21,8 +22,18 @@ public Handler(IPushService pushService, IUserContext userContext)
 
     public async Task<Unit> Handle(UpdateDeviceRegistrationCommand request, CancellationToken cancellationToken)
     {
-        var deviceRegistration = new DeviceRegistration(request.Platform, request.Handle, _activeDevice);
-        await _pushService.RegisterDeviceAsync(_activeIdentity, deviceRegistration);
+        var handle = PnsHandle.Parse(request.Handle, DeserializePlatform(request.Platform)).Value;
+        await _pushService.UpdateRegistration(_activeIdentity, _activeDevice, handle);
         return Unit.Value;
     }
+
+    private static PushNotificationPlatform DeserializePlatform(string platform)
+    {
+        return platform switch
+        {
+            "fcm" => PushNotificationPlatform.Fcm,
+            "apns" => PushNotificationPlatform.Apns,
+            _ => throw new NotImplementedException($"The platform {platform} is invalid.")
```

Comment:
You should always put quotes around variables in error messages, especially if those variables are of type string. Because imagine the value is an empty string. In this case, the error message would be "The platform  is invalid."

```suggestion
            _ => throw new NotImplementedException($"The platform '{platform}' is invalid.")
```